### DISCUSSION TITLE
App Router vs Pages Router Architecture & Differences?

### DISCUSSION THREAD

**Discussion by coltonehrman** [vercel/next.js#55533] [App Router]
It seems that these 2 modes operate differently at a fundamental level. One thing I noticed was that the App router (with custom server and in dev mode), seems to run a separate process for RSC and upon page execution. Is that the case?

Is there a high-level diagram that shows the internal architecture of Next.js and the different processes and runtimes that get executed and how a request/response lifecycle flows through the system?

**Comment by BradleyHsu** [comment]
+1, would love to better understand the internal architecture
